packages/s3-store/index.ts

```diff
@@ -25,6 +28,7 @@ type Options = {
   // but may increase it to not exceed the S3 10K parts limit.
   partSize?: number
   useTags?: boolean
+  maxConcurrentPartUploads?: number
```

Comment:
Do we need to expose this as an option? I think this falls under the category of we as upload experts should determine what is best for users for this low-level option.

Reply:
It all depends on the upload bandwidth of where the server is running, there is no magic value to be fair.
At the moment we are simply limiting how many parallel chunks and uploads can be processed at any point in time. 

For example, if you are running the server in a local network with a local minio, this value could be much higher since the upload throughput is higher.

Reply:
similarly, you run the uploads within the AWS platform (ECS/ECS, etc..) against S3, the upload would be faster than if you upload it from a server in a different provider. Meaning the concurrentPartUploads could increased.

My default value of max 30 concurrent chunks with a preferred size of 8MB means that the upload bandwidth will be around 240MB/s - the file system will match that with backpressure. 

Without backpressure, the file system could write parts at a speed of 1GB/s but upload at much slower rate

Reply:
That would be a bit too naive to not expose it. Perhaps we want to write some docs about the considerations, as I do think tus-node-server is also popular under front-end/full-stack devs with less awareness of this. 

how about something like this in the s3-store readme:

### Example: using `partSize` and `maxConcurrentParts` efficiently

These settings depend on the machine's processing power, memory, network bandwidth, and disk I/O capabilities.
Below are some recommendations to take into account but exact settings depend on your setup.

- **Uploading within the AWS platform to S3**. Given the high-speed environment and assuming robust server resources both values can be set higher than the default.
- **Uploading from outside the AWS platform to S3**.  These are variable-speed environments with more network fluctuations. The default `partSize` of 8 MiB and default `maxConcurrentParts` of 30 could work well for this (or lower on a resource-limited machine).
- **Using Local MinIO** (Local Network or On-Premises). Typically high-speed and low-latency but this should be balanced against the capabilities of your local server infrastructure. Values similar to or higher than those used within AWS might be suitable.

In general, the tradeoff is reliability vs. speed. A bigger `partSize` requires less roundtrips but is more prone to failure in less stable connections. Higher concurrency (`maxConcurrentParts`) increases CPU and memory usage.